Multi-GPU Kineto profiler test

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51391 Multi-GPU Kineto profiler test**

Summary:
Adding a test to check the kineto profiler on multiple gpus

Test Plan:
python test/test_profiler.py

Differential Revision: [D26160788](https://our.internmc.facebook.com/intern/diff/D26160788)

Commit:
Multi-GPU Kineto profiler test

Summary:
Adding a test to check the kineto profiler on multiple gpus

Test Plan:
python test/test_profiler.py

[ghstack-poisoned]

ngimel commented on test/test_profiler.py:
```diff
@@ -134,6 +135,31 @@ def test_kineto(self):
+    @unittest.skipIf(not kineto_available(), "Kineto is required")
+    @unittest.skipIf(not TEST_MULTIGPU, "Multiple GPUs needed")
+    def test_kineto_multigpu(self):
+        with profile(use_cuda=True, use_kineto=True) as prof:
```
given that we want to deprecate torch.autograd.profiler, we should switch to torch.profiler.profile here?

Commit:
Update on "Multi-GPU Kineto profiler test"


Summary:
Adding a test to check the kineto profiler on multiple gpus

Test Plan:
python test/test_profiler.py

Differential Revision: [D26160788](https://our.internmc.facebook.com/intern/diff/D26160788)

[ghstack-poisoned]
